src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,115 @@
+        }
+        return allManufacturers;
+    }
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertManufacturerRequest = "INSERT INTO manufacturers(name, country) values(?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement createManufacturerStatement = connection
+                         .prepareStatement(insertManufacturerRequest,
+                                 Statement.RETURN_GENERATED_KEYS)) {
+            createManufacturerStatement.setString(1, manufacturer.getName());
+            createManufacturerStatement.setString(2, manufacturer.getCountry());
+            createManufacturerStatement.executeUpdate();
+            ResultSet generatedKeys = createManufacturerStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to insert manufacturer in DB", e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE manufacturers SET is_deleted = true where id = ? ";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteManufacturerStatement = connection
+                        .prepareStatement(deleteRequest)) {
+            deleteManufacturerStatement.setLong(1, id);
+            return deleteManufacturerStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to delete manufacturer in DB", e);
+        }
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String getManufacturerRequest = "SELECT * FROM manufacturers "
+                                                + "WHERE id = (?) AND is_deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturerStatement = connection
+                        .prepareStatement(getManufacturerRequest)) {
+            getManufacturerStatement.setLong(1, id);
+            ResultSet resultSet = getManufacturerStatement.executeQuery();
+            Manufacturer manufacturer = new Manufacturer();
+            while (resultSet.next()) {
+                manufacturer.setId(resultSet.getObject("id", Long.class));
+                manufacturer.setName(resultSet.getString("name"));
+                manufacturer.setCountry(resultSet.getString("country"));
+                return Optional.of(manufacturer);
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to get manufacturer from DB", e);
```

Comment:
Use informative messages for exceptions.
```suggestion
            throw new DataProcessingException("Unable to get manufacturer from DB with id: "+id, e);
```